framework/topology.cpp

```diff
@@ -722,6 +730,14 @@ void apply_cpuset_param(char *param)
                 exit(EX_USAGE);
             }
             add_to_set(*cpu);
+        } else if ( strcmp( p.data(), "odd") == 0 || strcmp( p.data(), "even") == 0){
+            for (struct cpu_info &cpu : old_cpu_info)
+            {
+                if ((strcmp(p.data(), "even") == 0) && take_cpu(cpu.cpu_number, EVEN))
+                    add_to_set(cpu);
+                if ((strcmp(p.data(), "odd") == 0) && take_cpu(cpu.cpu_number, ODD))
+                    add_to_set(cpu);
+            } 
```

Comment:
`strcmp` is marked in glibc as being a pure function and the compiler is probably smart enough to realise that the comparison results can't change, but why not help it along the way?
```c++
    int desired_remainder = strcmp(p.data(), "odd") == 0 ? 1 : 0;
    for (struct cpu_info &cpu : old_cpu_info) {
        if ((cpu.cpu_number % 2) == desired_remainder)
            add_to_set(cpu);
    }
```